### DISCUSSION TITLE
The best way to model sentence responses with varying sentence lengths

### DISCUSSION THREAD

**Discussion by alireza-kr** [cvnlab/GLMsingle#147] [Q&A] [answered]
Hello

I have fMRI data from an experiment where subjects were presented with sentences from different categories (math and non-math) and were required to judge their truth or falsity. The sentences were presented word by word, with each word displayed for 500 ms. The sentences vary in length, averaging around seven words, but some contain up to 16 words. Consequently, the presentation time ranges from 2 to 8 seconds.

I want to model the response for each sentence, not each word, but I'm unsure of the best approach due to the varying sentence lengths. I can't use GLMsingle for this purpose.

Could you please provide your feedback on the best way to model the sentence responses? Is it better to standardize the sentence length to an average duration (e.g., 3.5 seconds), or should I model each word individually, label them under the same condition, and then average them?

Thanks!

**Comment by kendrickkay** [comment]
The choice of how to proceed depends on how you want to model the brain responses...  In other words, it depends on your theory of how the brain responses are generated.  For example, maybe you think that the main brain response of interest is triggered only at the end of the sentence.  Or, maybe you want to derive low-level "word-related" responses.  Or, maybe you seek to just get the overall BOLD response to each sentence.  In this latter case, it is still tricky because you have to compare timecourses that have different shapes.  Some strategies for dealing with different stimulus durations is discussed on the FAQ (https://glmsingle.readthedocs.io/en/latest/)

**Comment by alireza-kr** [comment]
Thank you, Kendrick. Yes, I read the FAQ before asking my question here, but I couldn't find a proper solution ... .

May I ask why you decided to impose the constraint of the same duration for all trials? I am considering whether it would be possible to modify some parts of the code to allow different durations for the trials.

**Comment by kendrickkay** [comment, marked as answer]
There are code architecture type reasons and also conceptual reasons for the choice.  As an example of the latter, there becomes an issue of how to interpret the beta magnitude for events of different durations.  For example, I suppose one could make the constraint that a stimulus of, say, 4 s has betas that are to be interpreted with respect to the betas that one would get from a 2-s stimulus, under the assumption that one expects the same response to two consecutive 2-s stimulus events (and expects the response to sum perfectly).  It is possible, but certainly would complicate the conceptual design...
